lib/instrumentation/index.js

```diff
@@ -102,7 +154,26 @@ Instrumentation.prototype.start = function () {
 
 Instrumentation.prototype._patchModule = function (exports, name, version, enabled) {
   this._agent.logger.debug('shimming %s@%s module', name, version)
-  return require('./modules/' + name)(exports, this._agent, version, enabled)
+  var patches = this._patches.get(name)
+  if (patches) {
+    for (let patch of patches) {
+      if (typeof patch === 'string') {
+        if (patch[0] === '.') {
+          patch = path.resolve(process.cwd(), patch)
+        }
+        patch = require(patch)
+      }
+
+      const type = typeof patch
+      if (type !== 'function') {
+        this._agent.logger.error('Invalid patch handler type:', type)
```

Comment:
Nit: If we really wanted to give a good error message here we should include `name`. But feel free to merge without this.

Reply:
I improve the message a bit.